src/sentry/static/sentry/app/views/admin/adminMail.tsx

```diff
@@ -5,13 +5,22 @@ import {t} from 'app/locale';
 import AsyncView from 'app/views/asyncView';
 import Button from 'app/components/button';
 
-export default class AdminEnvironment extends AsyncView {
-  getEndpoints() {
-    return [['data', this.getEndpoint()]];
-  }
-
-  getEndpoint() {
-    return '/internal/mail/';
+type Data = {
+  mailHost: string;
+  mailPassword: string;
+  mailUsername: string;
+  mailPort: string;
+  mailUseTls: string;
+  mailFrom: string;
+  mailListNamespace: string;
+  testMailEmail: string;
+};
+
+type State = AsyncView['state'] & {data: Data};
+
+export default class AdminEnvironment extends AsyncView<{}, State> {
```

Comment:
```suggestion
export default class AdminEnvironment extends AsyncView<AsyncView['props'], State> {
```

Reply:
I think I'll go back and fix these all at once in all of them